tests/CombinedRegexpTest.php

```diff
@@ -0,0 +1,255 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Strings\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Yiisoft\Strings\CombinedRegexp;
+
+final class CombinedRegexpTest extends TestCase
+{
+    /**
+     * @dataProvider dataMatchAny
+     */
+    public function testMatchAny(array $patterns, string $string, bool $expectedResult): void
+    {
+        $regexp = new CombinedRegexp($patterns);
+        $actualResult = $regexp->matches($string);
+        $message = sprintf(
+            'Failed to assert that string "%s" matches the string "%s".',
+            $regexp->getCompiledPattern(),
+            $string
+        );
+        $this->assertEquals($expectedResult, $actualResult, $message);
```

Comment:
Could be extracted to `private assertMatchesPattern(string $expected, string $actual, string $pattern)`